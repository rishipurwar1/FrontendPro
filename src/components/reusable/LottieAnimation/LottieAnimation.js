import { useLottie } from "lottie-react"

// const style = {
//   height: 420,
//   width: 420,
// }

export const LottieAnimation = ({
  animationDataFile,
  height = 320,
  width = 420,
  ...styles
}) => {
  const options = {
    animationData: animationDataFile,
    loop: true,
    autoplay: true,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice",
    },
  }

  const { View } = useLottie(options, { height, width, ...styles })

  return View
}
