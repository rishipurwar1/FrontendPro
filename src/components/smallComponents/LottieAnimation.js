import React, { useEffect, useState } from "react"
import Lottie from "react-lottie"
import PropTypes from "prop-types"
import { generateLottieOptions } from "../../utils"

export const LottieAnimation = React.memo(
  ({ animationDataFile, height = 420, width = 420 }) => {
    const [options, setOptions] = useState(null)
    useEffect(() => {
      animationDataFile &&
        setOptions(generateLottieOptions({ animationData: animationDataFile }))
    }, [animationDataFile])

    return (
      <>{options ? <Lottie height={height} width={width} options={options} /> : null}</>
    )
  }
)

LottieAnimation.propTypes = {
  animationDataFile: PropTypes.object.isRequired,
  height: PropTypes.number,
  weight: PropTypes.number,
}

LottieAnimation.displayName = "LottieAnimation"
