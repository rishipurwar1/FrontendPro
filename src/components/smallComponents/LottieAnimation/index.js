export { LottieAnimation as default } from "./LottieAnimation"
