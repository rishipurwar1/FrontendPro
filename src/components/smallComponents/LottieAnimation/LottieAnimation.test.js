import { render } from "@testing-library/react"
import { LottieAnimation } from "./LottieAnimation"
import rocketLoader from "../../../assets/animated_illustrations/loader.json"

describe("<LottieAnimation />", () => {
  it("should render the component properly", () => {
    const { container } = render(<LottieAnimation animationDataFile={rocketLoader} />)
    expect(container).toMatchSnapshot()
  })

  it("should render nothing if animation file is not passed", () => {
    const { container } = render(<LottieAnimation animationDataFile={null} />)
    expect(container.firstChild).toBeNull()
  })
})
