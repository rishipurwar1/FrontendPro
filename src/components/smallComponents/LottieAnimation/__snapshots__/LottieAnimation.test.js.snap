// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LottieAnimation /> should render the component properly 1`] = `
<div>
  <div
    aria-label="animation"
    role="button"
    style="width: 420px; height: 420px; overflow: hidden; margin: 0px auto; outline: none;"
    tabindex="0"
    title=""
  >
    <svg
      height="50"
      preserveAspectRatio="xMidYMid slice"
      style="width: 100%; height: 100%; transform: translate3d(0,0,0);"
      viewBox="0 0 50 50"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <clippath
          id="__lottie_element_2"
        >
          <rect
            height="50"
            width="50"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <g
        clip-path="url(#__lottie_element_2)"
      >
        <g
          opacity="1"
          style="display: block;"
          transform="matrix(1,0,0,1,0,0)"
        >
          <g
            opacity="1"
            transform="matrix(1,0,0,1,12.843000411987305,37.17100143432617)"
          >
            <path
              d=" M-2.6670000553131104,-5.281000137329102 C-2.6670000553131104,-5.281000137329102 -4.815999984741211,-1.6649999618530273 -4.815999984741211,-1.6649999618530273 C-4.815999984741211,-1.6649999618530273 -5.293000221252441,-0.8230000138282776 -2.242000102996826,2.2300000190734863 C0.8100000023841858,5.28000020980835 1.6540000438690186,4.801000118255615 1.6540000438690186,4.801000118255615 C1.6540000438690186,4.801000118255615 5.291999816894531,2.6480000019073486 5.291999816894531,2.6480000019073486"
              fill-opacity="0"
              stroke="rgb(104,38,235)"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-opacity="1"
              stroke-width="2"
            />
          </g>
          <g
            opacity="1"
            transform="matrix(1,0,0,1,25.084999084472656,39)"
          >
            <path
              d=" M3.9149999618530273,-7 C3.9149999618530273,-7 3.9149999618530273,2 3.9149999618530273,2 C3.9149999618530273,2 -2.0850000381469727,7 -2.0850000381469727,7 C-2.0850000381469727,7 -3.9149999618530273,-0.3179999887943268 -3.9149999618530273,-0.3179999887943268"
              fill-opacity="0"
              stroke="rgb(104,38,235)"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-opacity="1"
              stroke-width="2"
            />
          </g>
          <g
            opacity="1"
            transform="matrix(1,0,0,1,11,24.915000915527344)"
          >
            <path
              d=" M7,-3.9149999618530273 C7,-3.9149999618530273 -2,-3.9149999618530273 -2,-3.9149999618530273 C-2,-3.9149999618530273 -7,2.0850000381469727 -7,2.0850000381469727 C-7,2.0850000381469727 0.3179999887943268,3.9149999618530273 0.3179999887943268,3.9149999618530273"
              fill-opacity="0"
              stroke="rgb(104,38,235)"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-opacity="1"
              stroke-width="2"
            />
          </g>
          <g
            opacity="1"
            transform="matrix(1,0,0,1,28.881000518798828,21.121000289916992)"
          >
            <path
              d=" M19.118999481201172,-19.1200008392334 C19.118999481201172,-19.1200008392334 11.9350004196167,-18.993999481201172 5.090000152587891,-15.586999893188477 C2.8929998874664307,-14.489999771118164 0.6930000185966492,-13.020000457763672 -1.2769999504089355,-11.050999641418457 C-7.01800012588501,-5.309999942779541 -18.606000900268555,9.420999526977539 -18.606000900268555,9.420999526977539 C-18.606000900268555,9.420999526977539 -19.118999481201172,11.083000183105469 -15.098999977111816,15.104000091552734 C-11.081000328063965,19.121000289916992 -9.418999671936035,18.60700035095215 -9.418999671936035,18.60700035095215 C-9.418999671936035,18.60700035095215 5.474999904632568,6.85099983215332 11.050000190734863,1.2769999504089355 C13.01099967956543,-0.6869999766349792 14.479999542236328,-2.884000062942505 15.574999809265137,-5.085000038146973 C18.966999053955078,-11.918999671936035 19.118999481201172,-19.1200008392334 19.118999481201172,-19.1200008392334z"
              fill-opacity="0"
              stroke="rgb(104,38,235)"
              stroke-linecap="round"
              stroke-linejoin="miter"
              stroke-miterlimit="10"
              stroke-opacity="1"
              stroke-width="2"
            />
          </g>
          <g
            opacity="1"
            transform="matrix(1,0,0,1,34,16)"
          >
            <path
              d=" M0,-4 C-2.2090001106262207,-4 -4,-2.2090001106262207 -4,0 C-4,2.2090001106262207 -2.2090001106262207,4 0,4 C2.2090001106262207,4 4,2.2090001106262207 4,0 C4,-2.2090001106262207 2.2090001106262207,-4 0,-4z"
              fill-opacity="0"
              stroke="rgb(104,38,235)"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-opacity="1"
              stroke-width="2"
            />
          </g>
        </g>
        <g
          opacity="1"
          style="display: block;"
          transform="matrix(1,0,0,1,0,0)"
        >
          <g
            opacity="1"
            transform="matrix(1,0,0,1,6.01200008392334,43.98899841308594)"
          >
            <path
              d=" M0.9879999756813049,-4.988999843597412 C-2.6059999465942383,-2.2269999980926514 -4.98799991607666,-3.989000082015991 -4.98799991607666,4.98799991607666 C3.98799991607666,4.98799991607666 2.6700000762939453,2.1689999103546143 4.98799991607666,-0.9890000224113464"
              fill-opacity="0"
              stroke="rgb(104,38,235)"
              stroke-linecap="butt"
              stroke-linejoin="miter"
              stroke-miterlimit="10"
              stroke-opacity="1"
              stroke-width="2"
            />
          </g>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
